workshop/content/lab1.5_kafka.md

```diff
@@ -0,0 +1,272 @@
+kafka-controller-manager-67cb856b5c-79kwz   1/1     Running   0          4m23s
+```
+
+Also verify that the eventing knative pods are running:
+
+```
+oc get pods -n knative-eventing
+```
+
+```
+NAME                                               READY   STATUS      RESTARTS   AGE
+eventing-controller-848bcbd4f9-7zz68               1/1     Running     0          3d13h
+eventing-webhook-78dcf96448-6568h                  1/1     Running     0          3d13h
+imc-controller-8559ff856b-2sdk6                    1/1     Running     0          3d13h
+imc-dispatcher-575c7fcd8d-lrpmt                    1/1     Running     0          3d13h
+mt-broker-controller-56857cccc5-h49sp              1/1     Running     0          3d13h
+mt-broker-filter-784b7db965-5ngkk                  1/1     Running     0          3d13h
+mt-broker-ingress-6b9f847866-bhk5w                 1/1     Running     0          3d13h
+sugar-controller-594784974b-rpvsm                  1/1     Running     0          3d13h
+```
+
+We can also see the various kinds of objects we will be using, starting first with the eventing sources.  Notice the `KafkaSource`.
+
+```
+oc api-resources --api-group='sources.knative.dev'
+```
+
+```
+NAME               SHORTNAMES   APIGROUP              NAMESPACED   KIND
+apiserversources                sources.knative.dev   true         ApiServerSource
+containersources                sources.knative.dev   true         ContainerSource
+kafkasources                    sources.knative.dev   true         KafkaSource
+pingsources                     sources.knative.dev   true         PingSource
+sinkbindings                    sources.knative.dev   true         SinkBinding
+```
+
+Next we can see the specific eventing/messaging resources.  Notice the `Channel`.
+
+```
+oc api-resources --api-group='messaging.knative.dev'
+```
+
+```
+NAME               SHORTNAMES   APIGROUP                NAMESPACED   KIND
+channels           ch           messaging.knative.dev   true         Channel
+inmemorychannels   imc          messaging.knative.dev   true         InMemoryChannel
+subscriptions      sub          messaging.knative.dev   true         Subscription
+```
+
+### Knative Channel
+Channels are the mechanism that actually forward events through the system, from the source (producer) to the sink (consumer). By default, all Knative Eventing API use the InMemoryChannel (imc), which is great to get started, but is not applicable for all use cases due to it's inability to persist messages. To get durability, we need to use other channels, such as GCP PubSub or Kafka.
+
+
+### Verify Channel
+Let's verify that we are configured to use the `KafkaChannel`.
+
+```
+oc get configmap config-default-kafka-channel -o yaml -n knative-eventing
+```
+
+You can see the default for `kafka` is `KafkaChannel`.  Here's the relevant snippet:
+
+```
+apiVersion: v1
+data:
+  default-ch-config: |
+    namespaceDefaults:
```

Comment:
I see this as `clusterDefault` instead of `namespaceDefault`.

Reply:
👍